src/main/java/org/primefaces/component/calendar/BaseCalendarRenderer.java

```diff
@@ -188,7 +188,7 @@ protected Date convertToLegacyDateAPI(FacesContext context, UICalendar calendar,
     }
 
     protected Temporal convertToJava8DateTimeAPI(FacesContext context, UICalendar calendar, Class<?> type, String submittedValue) {
-        if (type == LocalDate.class || type == YearMonth.class) {
+        if (type == LocalDate.class || type == YearMonth.class || (type == LocalDateTime.class && !calendar.hasTime())) {
```

Comment:
Hm. Maybe we need to be cautious.
hasTime() looks at the pattern. DatePicker does not require to add HH:mm to pattern for LocalDateTime. You simply  do showTime=true.

Reply:
No DatePickerBase overrides it for DatePicker and it checks the `showTime` prop
```java
    @Override
    public boolean hasTime() {
        return this.isShowTime() || this.isTimeOnly();
    }
```

Reply:
Now a combination that would not would is LocalDateTime with timeOnly="true" but to me that is right not to make sense.  As you wouldn't set Month Day Year to 0/0/0 to give it a time.

Reply:
We added some magic to this for 10. Not totally sure how this reacts in all possible usecases. Did you check them?

Reply:
Thanks for checking! (I‘m currently on my iPad —> limited ability to check code)
I think we are safe. 😀 
I also agree LocalDateTime combined with timeOnly=true does not make sense.

Reply:
I have tested many scenarios and I think this change is a good one and now in the right spot!  Unit tests all passing as well.